### PR TITLE
Filter modal editor tests

### PR DESCRIPTION
**Pull request by ranquild**
Jest tests for "Editor" components in the `FilterModal`.

### CODE REVIEW

File: frontend/src/metabase/querying/components/FilterModal/TimeFilterEditor/TimeFilterEditor.unit.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+      userEvent.click(screen.getByText("is empty"));
+      userEvent.click(await screen.findByText("Not empty"));
+
+      expect(getNextFilterName()).toBe("Time is not empty");
+    });
+
+    it("should preserve values when switching operators", async () => {
+      const { query, stageIndex, column, filter } = createQueryWithFilter({
+        operator: "<",
+        values: [new Date(2020, 0, 1, 10, 20)],
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/components/FilterModal/TimeFilterEditor/TimeFilterEditor.unit.spec.tsx:201]: nit: I'm not a fan of using new Date, but maybe in tests it's fine

File: frontend/src/metabase/querying/components/FilterModal/TimeFilterEditor/TimeFilterEditor.unit.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        stageIndex,
+        column,
+        filter,
+      });
+
+      userEvent.clear(screen.getByDisplayValue("10:20"));
+      userEvent.type(screen.getByPlaceholderText("Min"), "11:40");
+      userEvent.clear(screen.getByDisplayValue("12:40"));
+      userEvent.type(screen.getByPlaceholderText("Max"), "15:10");
+      userEvent.click(document.body);
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/components/FilterModal/TimeFilterEditor/TimeFilterEditor.unit.spec.tsx:175]: if you need to trigger a blur event, there should be enough to use `userEvent.tab()`

image

File: frontend/src/metabase/querying/components/FilterModal/TimeFilterEditor/TimeFilterEditor.unit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+        filter,
+      });
+
+      userEvent.clear(screen.getByDisplayValue("10:20"));
+      userEvent.type(screen.getByPlaceholderText("Enter a time"), "11:40");
+      userEvent.click(document.body);
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/components/FilterModal/TimeFilterEditor/TimeFilterEditor.unit.spec.tsx:154]:
```suggestion
      userEvent.tab();
```

File: frontend/src/metabase/querying/components/FilterModal/CoordinateFilterEditor/CoordinateFilterEditor.unit.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+      userEvent.click(screen.getByText("between"));
+      userEvent.click(await screen.findByText("Greater than"));
+      expect(getNextFilterName()).toBeNull();
+
+      userEvent.type(screen.getByPlaceholderText("Enter a number"), "10");
+      userEvent.clear(screen.getByPlaceholderText("Enter a number"));
+      expect(getNextFilterName()).toBeNull();
+    });
+
+    it("should coerce invalid filter values", async () => {
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/components/FilterModal/CoordinateFilterEditor/CoordinateFilterEditor.unit.spec.tsx:134]: no await here
```suggestion
    it("should coerce invalid filter values", () => {
```

File: frontend/src/metabase/querying/components/FilterModal/CoordinateFilterEditor/CoordinateFilterEditor.unit.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+      userEvent.click(screen.getByText("between"));
+      userEvent.click(await screen.findByText("Greater than"));
+      userEvent.type(screen.getByPlaceholderText("Enter a number"), "20");
+      userEvent.click(document.body);
+
+      expect(getNextFilterName()).toBe("Latitude is greater than 20");
+      expect(onInput).toHaveBeenCalled();
+    });
+
+    it("should add a filter with two values", async () => {
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/components/FilterModal/CoordinateFilterEditor/CoordinateFilterEditor.unit.spec.tsx:82]:
```suggestion
    it("should add a filter with two values", () => {
```

File: frontend/src/metabase/querying/components/FilterModal/BooleanFilterEditor/BooleanFilterEditor.unit.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+      expect(getNextFilterName()).toBe("Is trial is not empty");
+      expect(screen.getByText("not empty")).toBeInTheDocument();
+      expect(
+        await screen.findByRole("checkbox", { name: "True" }),
+      ).not.toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "False" })).not.toBeChecked();
+
+      userEvent.click(screen.getByText("not empty"));
+      userEvent.click(await screen.findByText("Is"));
+      userEvent.click(await screen.findByRole("checkbox", { name: "True" }));
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/components/FilterModal/BooleanFilterEditor/BooleanFilterEditor.unit.spec.tsx:176]: I remember we discussed `findByRole` that it's slow, can't you use `byLabelText`?

File: frontend/src/metabase/querying/components/FilterModal/BooleanFilterEditor/BooleanFilterEditor.unit.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+    });
+
+    it("should allow to remove a filter", () => {
+      const { getNextFilterName } = setup({
+        query,
+        stageIndex,
+        column,
+      });
+
+      userEvent.click(screen.getByRole("checkbox", { name: "True" }));
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/components/FilterModal/BooleanFilterEditor/BooleanFilterEditor.unit.spec.tsx:115]: we need to verify that the value is `True` and then that it's `False` after uncheck for both checkboxes

**Review by uladzimirdev — COMMENTED** [review]
LGTM, minor comments left. 

I'd replace `userEvent.click(document.body)` with `userEvent.tab()` everywhere.

UPD: not everywhere, I see some tests failed if I use `userEvent.tab()` instead of click(document.body)

File: frontend/src/metabase/querying/components/FilterModal/BooleanFilterEditor/BooleanFilterEditor.unit.spec.tsx (hunk 6 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterModal/BooleanFilterEditor/BooleanFilterEditor.unit.spec.tsx:176]: Not unless `aria-label` is set explicitly. It seems there is no other way to do it without adding it.

File: frontend/src/metabase/querying/components/FilterModal/BooleanFilterEditor/BooleanFilterEditor.unit.spec.tsx (hunk 7 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterModal/BooleanFilterEditor/BooleanFilterEditor.unit.spec.tsx]: Fixed

File: frontend/src/metabase/querying/components/FilterModal/TimeFilterEditor/TimeFilterEditor.unit.spec.tsx (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterModal/TimeFilterEditor/TimeFilterEditor.unit.spec.tsx]: @uladzimirdev I've changed all document clicks with `userEvent.tab()` where it won't break the test

**Review by uladzimirdev — APPROVED** [review]

**Pull request merged**